src/dmd/target.d

```diff
@@ -774,6 +776,37 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+
+    /**
+     * Get targetInfo by key
+     * Params:
+     *  name = name of targetInfo to get
+     *  loc = location to use for error messages
+     * Returns:
+     *  Expression for the requested targetInfo
+     */
+    extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
+    {
+        switch (name.toDString)
+        {
+            case "cppRuntimeLibrary":
+                return new StringExp(loc, cast(void*)global.params.mscrtlib, global.params.mscrtlib ? strlen(global.params.mscrtlib) : 0);
+            default:
+                return new ErrorExp();
+        }
+    }
+
+    /**
+     * Get all supported target-info keys
+     * Returns:
+     *  Array of targetInfo keys
+     */
+    extern (C++) static const(char*)* allTargetInfos(size_t* numTargetInfos)
+    {
+        __gshared immutable(char*[]) keys = [ "cppRuntimeLibrary" ];
```

Comment:
As I mentioned in the other PR, I think this should be an enum which can be used by the above function.